test/gateways/authorize_net_test.exs

```diff
@@ -26,90 +26,120 @@ defmodule Gringotts.Gateways.AuthorizeNetTest do
   @bad_amount "a"
 
   @opts [
-    config: %{name: "64jKa6NA", transactionKey: "4vmE338dQmAN6m7B"},
-    refId: "123456",
-    order: %{invoiceNumber: "INV-12345", description: "Product Description"}, 
-    lineItem: %{itemId: "1", name: "vase", description: "Cannes logo", quantity: "18", unitPrice: "45.00"}
+    config: %{name: "64jKa6NA", transaction_key: "4vmE338dQmAN6m7B"},
+    ref_id: "123456",
+    order: %{invoice_number: "INV-12345", description: "Product Description"}, 
+    lineitems: %{
+      item_id: "1",
+      name: "vase",
+      description: "Cannes logo", 
+      quantity: "18", 
+      unit_price: "45.00"
+    }
   ]
   @opts_refund [
     config: %{name: "64jKa6NA", transaction_key: "4vmE338dQmAN6m7B"}, 
     ref_id: "123456", 
     payment: %{card: %{number: "5424000000000015", year: 2020, month: 12}}
   ]
-  @opts_refund_bad_payment [
+  @opts_store [
     config: %{name: "64jKa6NA", transaction_key: "4vmE338dQmAN6m7B"}, 
-    ref_id: "123456", 
+    profile: %{
+      merchant_customer_id: "123456", 
+      description: "Profile description here", 
+      email: "customer-profile-email@here.com"
+    },
+    customer_type: "individual",
+    validation_mode: "testMode"
+  ]
+  @opts_store_no_profile [
+    config: %{name: "64jKa6NA", transaction_key: "4vmE338dQmAN6m7B"}, 
+  ]
+  @opts_refund [
+    config: %{name: "64jKa6NA", transaction_key: "4vmE338dQmAN6m7B"}, 
+    ref_id: "123456",
+    payment: %{card: %{number: "5424000000000015", year: 2020, month: 12}}
+  ]
+  @opts_refund_bad_payment [
+    config: %{name: "64jKa6NA", transaction_key: "4vmE338dQmAN6m7B"},
+    ref_id: "123456",
     payment: %{card: %{number: "123", year: 2020, month: 12}}
   ]
   @opts_store [
-    config: %{name: "64jKa6NA", transaction_key: "4vmE338dQmAN6m7B"}, 
-    profile: %{merchant_customer_id: "123456", description: "Profile description here", email: "customer-profile-email@here.com"}
+    config: %{name: "64jKa6NA", transaction_key: "4vmE338dQmAN6m7B"},
+    profile: %{merchant_customer_id: "123456",
+      description: "Profile description here", 
+      email: "customer-profile-email@here.com"
+    }
   ]
   @opts_store_no_profile [
     config: %{name: "64jKa6NA", transaction_key: "4vmE338dQmAN6m7B"}, 
   ]
-
+  @opts_customer_profile [
+    config: %{name: "64jKa6NA", transaction_key: "4vmE338dQmAN6m7B"}, 
+    customer_profile_id: "1814012002",
+    validation_mode: "testMode"    
+  ]
+  
```

Comment:
since you are repeating `config: %{name: "64jKa6NA", transaction_key: "4vmE338dQmAN6m7B"}` in every options module attribute, make it as required module attribute 
`@req_opts [config: %{name: "64jKa6NA", transaction_key: "4vmE338dQmAN6m7B"}]`

and then you can use it like this

`ANet.authorize(@amount, @card, @req_opts ++ @opts)`

Reply:
Or you could just extract the map out and bind it to an `@auth` attr, then replace all occurrences of the map with `@auth`, no need of `++`.